### DISCUSSION TITLE
Window combi text misaligned

### DISCUSSION THREAD

**Discussion by gavsiu** [davatorium/rofi#1435] [Theming] [answered]
### Rofi version (rofi -v)

1.7.0

### Configuration

https://gist.github.com/gavsiu/dcd533ab26d77edf7ad4f2cc5f946901

### Launch command

rofi -show combi -modi combi,drun,monitors:/home/gavsiu/.config/rofi/scripts/monitor_layout.sh,ssh

### Step to reproduce

Launch rofi in combi mode with window.

### Expected behavior

![2021-09-05-110225_4480x1440_scrot](https://user-images.githubusercontent.com/806613/132137100-0019dd3e-9498-4b85-ae65-900914930cbb.png)
rofi 1.6.1 with config/theme updated for 1.7.0.

### Actual behavior

![2021-09-05-110035_4480x1440_scrot](https://user-images.githubusercontent.com/806613/132137135-c3e47e3a-fa49-44b7-8ce0-3355159211ca.png)

### Additional information

My theme is just pretty much a modified Arc-Dark. Same issue is visible if I just switch to Arc-Dark or seems like any of the old themes.

**Comment by DaveDavenport** [comment, marked as answer]
Add this:

```css
* {
	font: "Iosevka 11";
}
```
after the @theme line in your config.

Next time please read pinned issue and follow the instruction in the template. (I missed the theme, so I could not test).

**Reply by gavsiu** [reply]
Sorry, I didn't realize this was just a theme issue considering the buttons/icons not being centred was considered a bug.

Also, that didn't work. Not for my theme nor Arc-Dark.

**Reply by DaveDavenport** [reply]
nothing changed in the rendering code and rofi does not do any active alignment (that is why you need a monospace font for listview to get it to work).
Can you please post your theme so I can test?

**Reply by DaveDavenport** [reply]
That one turned out to be an actual bug, this seems to be a theme issue, if it turns out to be a bug I will turn it into an issue (see pinned isssue)

**Reply by gavsiu** [reply]
[colors.rasi](https://gist.github.com/gavsiu/f0339c05d5e77d0020c1bf9f8de2c033)
[theme.rasi](https://gist.github.com/gavsiu/ffbbb36875155957c9bcd2f772a0c8da)

Arc-Dark and the other themes I tried are from your repo.

**Reply by DaveDavenport** [reply]
thanks.

**Reply by DaveDavenport** [reply]
Can you try: 
```bash
rofi -show window -theme-str 'element-text { font: "mono 18"; }'
```

it looks to me that your font is not monospaced, this fixes it for me.

**Reply by DaveDavenport** [reply]
(not sure why it works on 1.6.1, it also fails there for me).

**Reply by gavsiu** [reply]
Looks the same running that, just bigger. The font is monospace only though. [Iosevka](https://github.com/be5invis/Iosevka).

**Reply by DaveDavenport** [reply]
weird, with the above line it works for me on both 1.6.1 and 1.7.0.  without monospace I get your issue.
Can you pastebin output of `rofi -dump-config`  in full?

(text rendering has not been changed between 1.6.1 and 1.7,0. Still uses pango)

**Reply by gavsiu** [reply]
It looks like the spacing between the desktop name and application name is exactly the same and not dynamic like before. Not sure how it looks for everyone else, but I'm using bspwm wm with btops to dynamically rename desktops.

**Reply by gavsiu** [reply]
https://gist.github.com/gavsiu/b9f193ea41039f6ab632f0cf133154f7

**Reply by DaveDavenport** [reply]
```window-format: "{w}    {c}   {t}";```
is the option that determines what is showed, this has not changed. 

Let me try on arch, as it works on ubuntu.

**Reply by gavsiu** [reply]
Thanks for your time.

**Reply by DaveDavenport** [reply]
![rofi-2021-06-23-0039-00000](https://user-images.githubusercontent.com/406245/132143342-716b74db-2c6c-45b8-8bc6-10e740d73b29.png)

the plot thickens.

**Reply by DaveDavenport** [reply]
![rofi-2021-06-23-0043-00001](https://user-images.githubusercontent.com/406245/132143422-77c054f1-6019-4b3d-af00-3d36f2d58b83.png)
with the right font.

**Reply by DaveDavenport** [reply]
I don't directly see the issue (as I cannot reproduce) , Ill give it another look tomorrow when rested :smile:

**Reply by DaveDavenport** [reply]
(we might be able to fix it by not relying on 'tabs' but use the advanced formatting in  `window-format` where you specify the pad length.)

**Reply by gavsiu** [reply]
Main difference between your screenshot and mine is your workspace is labeled "Workspace 1" with 11 characters. My workspaces have different lengths. It looks like the spacing between the app name and window title is dynamic, but the spacing between the workspace name and app name isn't anymore.

**Reply by DaveDavenport** [reply]
I am pretty sure this code has not changed.  The default config as *I* ship him relied on tabs and did nothing fancy.

**Reply by DaveDavenport** [reply]
anyway it is late I will double check tomorrow, bugs are always possible,

**Reply by DaveDavenport** [reply]
sleepy brain working slow, this patch went in, might have broken it: https://github.com/davatorium/rofi/pull/1288

**Reply by DaveDavenport** [reply]
you where absolutely right.

**Reply by DaveDavenport** [reply]
On I3 (that has pango-markup in desktop names) the problem does not exists, thats why I could not reproduce...

**Reply by DaveDavenport** [reply]
It has been fixed, so Ill mark it as answered.
